src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -125,6 +125,9 @@ bool DoorLockServer::SetLockState(chip::EndpointId endpointId, DlLockState newLo
     // Schedule auto-relocking
     if (success && LockOperationTypeEnum::kUnlock == opType)
     {
+        // Reset code wrong entry attempts (in case there were any incorrect wrong code retries prior) since unlock was a success.
+        ResetWrongCodeEntryAttempts(endpointId);
```

Comment:
Shouldn't we only do that in cases when credentials were involved?  If this was an unlock without credentials presented, it shouldn't reset the wrong-credential attempt counter....

Reply:
agreed. I missed that. i have to check if credential is pincode and hasValue here too.

Reply:
just checking if credential is not null and not empty as dicussed. done